analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,248 @@
+                {
+                    programState = InvocationExpressionPreProcess(invocation, programState);
+                }
+
+                return base.PreProcessInstruction(programPoint, programState);
+            }
+
+            private ProgramState ObjectCreationPostProcess(ObjectCreationExpressionSyntax objectCreation, ProgramState programState)
+            {
+                if (semanticModel.GetSymbolInfo(objectCreation).Symbol is {} symbol
+                    && symbol.ContainingType is {} symbolType
+                    && symbolType.IsAny(VulnerableTypes))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.NotInitialized);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.NotInitialized);
+                }
+
+                return programState;
+            }
+
+            private ProgramState InvocationExpressionPostProcess(InvocationExpressionSyntax invocation, ProgramState programState)
+            {
+                if (IsCreateMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.NotInitialized);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.NotInitialized);
+                }
+                else if (IsGenerateKeyMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = GetSymbolicValue(invocation, programState);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.Initialized);
+                }
+                else if (IsGenerateIVMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = GetSymbolicValue(invocation, programState);
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.Initialized);
+                }
+                else if (IsRngCryptoServiceProviderSanitizer(invocation, semanticModel)
+                         && semanticModel.GetSymbolInfo(invocation.ArgumentList.Arguments[0].Expression).Symbol is {} symbol
+                         && symbol.GetSymbolType().Is(KnownType.System_Byte_Array)
+                         && symbol.HasConstraint(ConstantByteArraySymbolicValueConstraint.Constant, programState))
+                {
+                    var symbolicValue = programState.GetSymbolValue(symbol);
+                    programState = programState.RemoveConstraint(symbolicValue, ConstantByteArraySymbolicValueConstraint.Constant);
```

Comment:
Shouldn't we mark the byte array as modified here?

Reply:
Currently it's fine either way since we just check for `Constant`